include/rocksdb/options.h

```diff
@@ -1338,6 +1338,11 @@ struct WriteOptions {
   // Default: false
   bool low_pri;
 
+  // If true, this writebatch will use its own insert hints in concurrent write
+  //
+  // Default: false
+  bool hint_per_batch;
```

Comment:
which of `memtable_insert_with_hint_prefix_extractor` and `hint_per_batch` should take precedence? It seems to me it should be `hint_per_batch` because its a per batch option which is more specific. And we should document the behavior in the inline comment.

Reply:
Should we by default enable it if `memtable_insert_with_hint_prefix_extractor` is set?
Extra option always makes it harder to maintain.

Reply:
The two options are not compatible with each other. `memtable_insert_with_hint_prefix_extractor` preserve the hint across different write batches, and `hint_per_batch` has nothing to do with prefixes. 

But I'm wondering whether `hint_per_batch` can always enable. Like for each write batch we keep the splice for the first key, then detect if its a sequential insert. If so, reuse the hint, otherwise discard the hint and start over.

Reply:
For now I only enable `hint_per_batch` when `concurrent_memtable_writes` is set to true, because   in non-concurrent write either `memtable_insert_with_hint_prefix_extractor` or `seq_splice_` will be used, which i think will make no much difference in performance compared to `hint_per_batch`.

Reply:
After some discussion we think having an extra `hint_per_batch` write option is more flexible and better fit our needs.